src/routes/ProfessionSubcategoryScreen.tsx

```diff
@@ -77,51 +77,36 @@ interface ProfessionSubcategoryScreenPropsType {
 
 const ProfessionSubcategoryScreen = ({ route, navigation }: ProfessionSubcategoryScreenPropsType): JSX.Element => {
   const { extraParams } = route.params
-  const { disciplineID, disciplineTitle } = extraParams
-  const [subcategories, setSubcategories] = useState<ProfessionSubcategoryType[]>([])
-  const [isLoading, setIsLoading] = useState<boolean>(true)
-  const [selectedId, setSelectedId] = useState<number | null>(null)
-  const [count, setCount] = useState<number>(0)
-  const [error, setError] = useState<string | null>(null)
+  const { module } = extraParams
 
-  useFocusEffect(
-    React.useCallback(() => {
-      const url = ENDPOINTS.subCategories.all.replace(':id', disciplineID.toString())
-      axios
-        .get(url)
-        .then(response => {
-          setSubcategories(response.data)
-          setCount(response.data.length)
-          setError(null)
-        })
-        .catch(e => {
-          setError(e.message)
-        })
-        .finally(() => {
-          setIsLoading(false)
-        })
-      setSelectedId(-1)
-    }, [disciplineID])
-  )
+  const [selectedId, setSelectedId] = useState<number | null>(null)
+  const { data: disciplines, error, loading } = useLoadDisciplines(module)
 
   const titleCOMP = (
     <Title>
       <>
-        <Text style={styles.screenTitle}>{disciplineTitle}</Text>
+        <Text style={styles.screenTitle}>{module?.title}</Text>
         <Text style={styles.description}>
-          {count} {count === 1 ? labels.home.unit : labels.home.units}
+          {module.numberOfChildren} {module.numberOfChildren === 1 ? labels.home.unit : labels.home.units}
         </Text>
       </>
     </Title>
   )
 
-  const Item = ({ item }: { item: ProfessionSubcategoryType }): JSX.Element | null => {
-    if (item.total_documents === 0) {
+  const ListItem = ({ item }: { item: DisciplineType }): JSX.Element | null => {
+    if (item.numberOfChildren === 0) {
       return null
     }
     const selected = item.id === selectedId
     const descriptionStyle = selected ? styles.clickedItemDescription : styles.description
     const badgeStyle = selected ? styles.clickedItemBadgeLabel : styles.badgeLabel
+    const description = module.isLeaf
+      ? item.numberOfChildren === 1
+        ? labels.home.word
+        : labels.home.words
+      : item.numberOfChildren === 1
+      ? labels.home.unit
+      : labels.home.units
```

Comment:
Indentation

Reply:
its prettier :unamused: 
made it a one-liner

Reply:
oke is also not working...